frontend/platform.bible-extension/src/main.ts

```diff
@@ -1,80 +1,264 @@
+      const name = (await pdp.getSetting('platform.name')) ?? projectId;
+      const dictionary = await pdp.getSetting('fw-lite-extension.fwDictionaryCode');
+      const options: OpenWebViewOptionsWithProjectId = { existingId: '?', projectId };
+      if (dictionary) {
+        logger.info(`Project '${name}' is using FieldWorks dictionary '${dictionary}'`);
+        urlHolder.dictionaryUrl = `${urlHolder.baseUrl}/paratext/fwdata/${dictionary}`;
+        await papi.webViews.openWebView(mainWebViewType, undefined, options);
+      } else {
+        logger.warn(`FieldWorks dictionary not selected for project '${name}'`);
+        await papi.webViews.openWebView(dictionarySelectWebViewType, { type: 'float' }, options);
+      }
+      return { success: true };
+    },
+  );
   const findEntryCommandPromise = papi.commands.registerCommand(
     'fwLiteExtension.findEntry',
-    (entry: string) => {
-      onFindEntryEmitter.emit({entry});
-      return {
-        success: true,
-      };
+    async (webViewId: string, word: string) => {
+      const projectId = await getProjectIdFromWebViewId(webViewId);
+      if (!projectId) return { success: false };
+      logger.info(
+        `Opening FieldWorks Lite find entry view for project '${projectId}' with word '${word}'`,
+      );
+      const options: WordWebViewOptions = { existingId: '?', projectId, word };
+      void papi.webViews.openWebView(findWordWebViewType, undefined, options);
+      return { success: true };
     },
   );
-  const simpleFindEntryCommandPromise = papi.commands.registerCommand(
-    'fwLiteExtension.simpleFind',
-    () => {
-      onFindEntryEmitter.emit({entry: 'apple'});
-      return {
-        success: true,
-      };
+  const openFwLiteCommandPromise = papi.commands.registerCommand(
+    'fwLiteExtension.openFWLite',
+    async (webViewId: string) => {
+      const projectId = webViewId
+        ? (await papi.webViews.getOpenWebViewDefinition(webViewId))?.projectId
+        : undefined;
+      const options: OpenWebViewOptionsWithProjectId = { existingId: '?', projectId };
+      void papi.webViews.openWebView(mainWebViewType, undefined, options);
+      return { success: true };
+    },
+  );
+  const selectFwDictionaryCommandPromise = papi.commands.registerCommand(
+    'fwLiteExtension.selectDictionary',
+    async (projectId: string, dictionaryCode: string) => {
+      logger.info(`Selecting FieldWorks dictionary '${dictionaryCode}' for project '${projectId}'`);
+      const pdp = await papi.projectDataProviders.get('platform.base', projectId);
+      await pdp.setSetting('fw-lite-extension.fwDictionaryCode', dictionaryCode);
+      return { success: true };
+    },
+  );
+  const fwDictionariesCommandPromise = papi.commands.registerCommand(
+    'fwLiteExtension.fwDictionaries',
+    async () => {
+      logger.info('Fetching local FieldWorks dictionaries');
+      return await fwLiteApi.getProjects();
```

Comment:
this is going to return all dictionaries, including harmony dictionaries, however the API calls currently have it hardcoded to only work for FwData. This is probably something I should change on our end and make a different API for fetching a list of projects, the way it works now kinda sucks.